modular_skyrat/master_files/code/modules/mob/living/simple_animal/friendly/poppy.dm

```diff
@@ -63,12 +76,36 @@
 	regenerate_icons()
 
 /mob/living/simple_animal/pet/poppy/Life(delta_time = SSMOBS_DT, times_fired)
-	if(buckled || client || !DT_PROB(0.5, delta_time))
-		return ..()
+	if(buckled || client)
+		return
+
+	if(pulledby)
+		set_resting(FALSE)
+
+	if(!near_engine)
+		for(var/obj/machinery/power/supermatter_crystal/sm_crystal in view(8, src))
```

Comment:
That is *way* too big of a range to be checking every like 4 seconds to then iterate over to find an SM crystal. It's a cute idea, but you should rather hook a signal upon Poppy changing area, and just check there if the new area is the Supermatter's area. Would be *much* less expensive.